Dockerfile

```diff
@@ -159,26 +135,29 @@ RUN AVAILABLE_TERRAFORM_VERSIONS="1.2.9 1.3.10 1.4.6 ${DEFAULT_TERRAFORM_VERSION
 # Creating the individual distro builds using targets
 FROM alpine:${ALPINE_TAG} AS alpine
 
-# atlantis user for gosu and OpenShift compatibility
+EXPOSE ${ATLANTIS_PORT:-4141}
+
+HEALTHCHECK --interval=5m --timeout=3s \
+  CMD curl -f http://localhost:${ATLANTIS_PORT:-4141}/healthz || exit 1
+
+# Set up the 'atlantis' user and adjust permissions
 RUN addgroup atlantis && \
     adduser -S -G atlantis atlantis && \
-    adduser atlantis root && \
     chown atlantis:root /home/atlantis/ && \
-    chmod g=u /home/atlantis/ && \
-    chmod g=u /etc/passwd
+    chmod u+rwx /home/atlantis/ && \
+    chmod u+rw /etc/passwd
 
```

Comment:
How come the chmod of /etc/passwd needs to change? If the line is removed, can atlantis still function?

Reply:
Right but I don't understand why `/etc/passwd` perms need to be changed. Can we get away with only changing `/home/atlantis` ?

Reply:
Ah, I didn't want to break anything. Therefor I just gave it read write permissions. We can't use `g=u` anymore though.

Reply:
no need to change the /etc/passwd file, Atlantis does not manage that or needs to.

Reply:
I think it might be managed in the docker-entrypoint.sh

https://github.com/runatlantis/atlantis/blob/a542aa8f2015e67957c5fdb6ec994080561aa1ec/docker-entrypoint.sh#L26-L28

Reply:
But why is doing this if useradd command will update /etc/passwd automatically?

Reply:
I just tested this : 

Dockerfile for test : 
```
ARG ALPINE_TAG=3.18.4
ARG DEBIAN_TAG=12.2-slim
FROM debian:${DEBIAN_TAG} as debian-base
#FROM alpine:${ALPINE_TAG} AS alpine
CMD ["/bin/sh", "-c", "echo 'It works!'"]
```

I built and image for alpine and debian

then in debian:

Reply:
and the other thing is that we should be setting the UID and GID to 100:1000 which is the default for a brand-new user when added to a new system.

Reply:
Yep very true. atlantis setup should not be screwing around with /etc/passwd.

I figured that could be addressed in a separate PR since this /etc/passwd modification already exists in atlantis for years, no?

Reply:
I think we should tackle all here since we are already removing the commands and doing security improvements

Reply:
I'm trying to understand the reason behind the old approach.
I can't find any reason why that was added.

Reply:
Ok but the more changes made, the more likely something can break. I'm all for more security, provided @bschaatsbergen has the appetite to keep going 🙏

Reply:
Is this really the case? I'm happy to add it, but couldn't find it anywhere. I know that users start from 1000 though. Is there a particular reason we should explicitly set this? Omitting it would prevent a uid and gid conflict right? @jamengual

Reply:
I've set uid/gid 100:1000 and removed the passwd entry lines.

Reply:
alpine and debian have different `uid`s it appears:  https://github.com/runatlantis/atlantis/pull/3317#discussion_r1384175827

Reply:
@arohter it's removed again, to stay inline with what's in the Dockerfile now. to avoid touching the uid and gid at all.